### ISSUE TITLE
asynchttpserver **Basic Usage** example crashes

### ISSUE THREAD

**Issue by shiftrightonce** [nim-lang/Nim#14516]
Function `echo` outputs the wrong string.

### Example
```nim
import asynchttpserver, asyncdispatch

var server = newAsyncHttpServer()
proc cb(req: Request) {.async.} =
   await req.respond(Http200, "Hello World")

waitFor server.serve(Port(9000), cb)
```

### Current Output
please check whether the problem still exists in git head before posting,
see [rebuilding the compiler](https://nim-lang.github.io/Nim/intern.html#rebuilding-the-compiler).
```
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

### Expected Output
```
Hello World
```

### Possible Solution

* In file httpcore.nim, line 180. Checking if the headers variable is nil before calling **hasKey** fix the issue

### Additional Information
If it's a regression, you can help us by identifying which version introduced
the bug, see [Bisecting for regressions](https://nim-lang.github.io/Nim/intern.html#bisecting-for-regressions),
or at least try known past releases (eg `choosenim 1.2.0`).

If it's a pre-existing compiler bug, see [Debugging the compiler](https://nim-lang.github.io/Nim/intern.html#debugging-the-compiler)
which should give more context on a compiler crash.

* Issue #abc is related, but different because of ...
* This issue is blocking my project xyz

```
$ nim -v
Nim Compiler Version 0.1.2
# make sure to include the git hash if not using a tagged release
```

**Comment by ghost** [comment]
That was fixed in devel and backported to 1.2.2 which will be released soon. Please close :)
https://github.com/nim-lang/Nim/issues/13866
https://github.com/nim-lang/Nim/pull/13867

**Comment by shiftrightonce** [comment]
Thanks a lot @Yardanico
